### DISCUSSION TITLE
Type annotations

### DISCUSSION THREAD

**Discussion by joouha** [MaartenGr/BERTopic#1796] [Ideas]
Hello,

Are you interested in having BERTopic fully type annotated (i.e. passing mypy checks)?

If so I'm willing to contribute this.

Let me know your thoughts!

Josiah

**Comment by MaartenGr** [comment]
Hi Josiah,

Thank you for your help! At the moment, I am not considering fully type annotated checks in BERTopic as I want to have a degree of freedom with respect to some of the values passed to certain variables. Having said that, there are many open issues that you can help out with if you want. For instance, https://github.com/MaartenGr/BERTopic/issues/1771 needs changes in order to use zero-shot modeling with other techniques. Aside from that, tutorials and other documentation changes are very helpful! 

Cheers,
Maarten
